### PR TITLE
Last quantile

### PR DESCRIPTION
**Pull request by Rlamboll**
# Pull request

Please confirm that this pull request has done the following:

- [x] Tests added
- [x] Documentation added (where applicable)
- [x] Example added (either to an existing notebook or as a new notebook, where applicable)
- [x] Description in ``CHANGELOG.rst`` added

## Adding to CHANGELOG.rst

Please add a single line in the changelog notes similar to one of the following:

```
- (`#XX <https://github.com/GranthamImperial/silicone/pull/XX>`_) Added feature which does something
- (`#XX <https://github.com/GranthamImperial/silicone/pull/XX>`_) Fixed bug identified in (`#YY <https://github.com/GranthamImperial/silicone/issues/YY>`_)
```

### CODE REVIEW

**Review by jkikstra — APPROVED** [review]
Great to see that you're extending the functionality of silicone - I think it will be very beneficial to have this additional infiller in the package.

I was able to run and it performs as expected - so all good from a functionality point-of-view.

However, I was reflecting a bit on that this 'infiller' is significantly different from the others and i was wondering if this should be reflected in the documentation a bit more.

All other infillers are about adding more variables, where this is about extending a variable temporally. What do you think about making this separation more clear? Would it be conceivable to that in the future more 'temporal-extension' infillers?

**Review by znicholls — COMMENTED** [review]
My two cents from a very quick read. Looks like awesome functionality, very nice. My only question: why the changes to the quantile walk infilled? As a note, if these changes to the quantile walk are kept then they would be API breaking changes so we'd need to bump the major version on next release. My suggestion would be to only make those changes if they're really needed (they may well be).

**Comment by Rlamboll** [comment]
Thanks for your reviews guys! I was indeed thinking of making a separate section for this (on the basis that there may be other projector functions upcoming) so if it occurs to other people then I'll make a separate section for them. @znicholls the changes to the quantile walk are simply to extract a part of the code to allow it to be reused - there is no change to the output (you will see I haven't changed the test for it) so it's not API breaking. The next push will feature additional quantile options that are tedious to duplicate, so it's better to have reused code.

File: src/silicone/database_crunchers/equal_quantile_walk.py [hunk 1]
```diff
@@ -61,19 +61,15 @@ def derive_relationship(self, variable_follower, variable_leaders):
         lead_ts = self._db.filter(variable=variable_leaders).timeseries()
         lead_unit = lead_ts.index.get_level_values("unit")[0]
 
-        def filler(in_iamdf, interpolate=False):
```
Comment (znicholls) [inline src/silicone/database_crunchers/equal_quantile_walk.py:64]: Removing `interpolate` as a keyword argument is minor but nonetheless API breaking (it is true that we never tested it).

**Comment by znicholls** [comment]
Definitely a good idea to reuse. I've highlighted the API breaking bit, it's pretty minor so not the end of the world.

**Comment by Rlamboll** [comment]
OK, I refactored it to separate out time projectors (I decided to make it the name sound less like a sci-fi body implant). @znicholls I don't see any inline comments, should I?

**Comment by jkikstra** [comment]
Looking good to me. 

I also can't see the comment.

**Review by znicholls — COMMENTED** [review]
Oops sorry i forgot to hit send

File: src/silicone/database_crunchers/equal_quantile_walk.py (hunk 1 above)
Comment (Rlamboll) [reply src/silicone/database_crunchers/equal_quantile_walk.py:64]: Oh, that. Yes, it was never actually attached to anything, so I hope no one used it and we should definitely get rid of it (I continue to think that people should just interpolate everything before they put the data in). We can spike the version number to 2 if you want as a result when we do the next release. Perhaps we should write a helper function that does the interpolation in a sensible way instead.

Comment (znicholls) [reply src/silicone/database_crunchers/equal_quantile_walk.py:64]: I think just putting something in the CHANGELOG will suffice (particularly given that we never tested it and probably no-one used it). It just jumped out so I thought I should raise.

> I continue to think that people should just interpolate everything before they put the data in

We could certainly add that advice in docs and/or add stronger checking.

Comment (Rlamboll) [reply src/silicone/database_crunchers/equal_quantile_walk.py:64]: OK, I'll update the changelog in the next push.

**Pull request merged**